File: Keyboard-Handler/src/SwitchKeyboard.m
#include <Carbon/Carbon.h>
#include <Foundation/Foundation.h>

int main (int argc, const char * argv[]) {
    if (argc != 2) {
        NSLog(@"Please provide positional arg: `SwitchKeyboard <layout>`");
        return -1;
    }
    NSString *layout = [NSString stringWithFormat:@"com.apple.keylayout.%s", argv[1]];
    NSArray* sources = CFBridgingRelease(TISCreateInputSourceList((__bridge CFDictionaryRef)@{ (__bridge NSString*)kTISPropertyInputSourceID : layout }, FALSE));
    TISInputSourceRef source = (__bridge TISInputSourceRef)sources[0];
    OSStatus status = TISSelectInputSource(source);
    if (status != noErr) {
        return -1;
    }
    return 0;
}

File: Keyboard-Handler/src/SwitchKeyboard.swift
import Carbon
import Foundation

let arguments = CommandLine.arguments
if arguments.count != 2 {
    print("Please provide positional arg: `SwitchKeyboard <layout>`")
    exit(-1)
}
let layout = "com.apple.keylayout.\(arguments[1])"
guard let sources = TISCreateInputSourceList([kTISPropertyInputSourceID: layout] as CFDictionary, false)?.takeRetainedValue() as? [TISInputSource] else {
    exit(-1)
}
let source = sources[0]
let status = TISSelectInputSource(source)
if status != noErr {
    exit(-1)
}
exit(0)
